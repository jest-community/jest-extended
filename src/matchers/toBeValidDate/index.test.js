import each from 'jest-each';

import matcher from './';

expect.extend(matcher);

describe('.toBeValidDate', () => {
  test('passes when given a valid date', () => {
    expect(new Date()).toBeValidDate();
  });

  test('fails when given an invalid date', () => {
    expect(() => expect(new Date('2018-90-90T13:00:00.000Z')).toBeValidDate()).toThrowErrorMatchingSnapshot();
  });

  test('fails when not given non-date values', () => {
    expect(() => expect(1).toBeValidDate()).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toBeValidDate', () => {
  each([
    [new Date('01/90/2018')],
    [new Date('32/01/2018')],
    [false],
    [true],
    [0],
    [''],
    [{}],
    [() => {}],
    [undefined],
    [null],
    [NaN]
  ]).test('passes when not given a date: %s', given => {
    expect(given).not.toBeValidDate();
  });

  test('fails when given a valid date value', () => {
    expect(() => expect(new Date('2018-01-01T13:00:00.000Z')).not.toBeValidDate()).toThrowErrorMatchingSnapshot();
  });
});
