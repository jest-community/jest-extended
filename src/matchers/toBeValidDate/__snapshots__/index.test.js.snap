// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeValidDate fails when given a valid date value 1`] = `
"<dim>expect(</><red>received</><dim>).not.toBeValidDate(</><dim>)</>

Expected value to not be a valid date received:
  <red>2018-01-01T13:00:00.000Z</>"
`;

exports[`.toBeValidDate fails when given an invalid date 1`] = `"Invalid time value"`;

exports[`.toBeValidDate fails when not given non-date values 1`] = `
"<dim>expect(</><red>received</><dim>).toBeValidDate(</><dim>)</>

Expected value to be a valid date received:
  <red>1</>"
`;
